pallets/utxo/src/lib.rs

```diff
@@ -386,7 +521,7 @@ pub mod pallet {
         trx.encode()
     }
 
-    fn disperse_reward<T: Config>(auths: &[H256], block_number: T::BlockNumber) {
+    fn reward_tx_validation<T: Config>(auths: &[H256], block_number: T::BlockNumber) {
         let reward = <RewardTotal<T>>::take();
         let share_value: Value =
             reward.checked_div(auths.len() as Value).ok_or("No authorities").unwrap();
```

Comment:
Besides the unwrap, where I explained how it should be treated in another comment.

I feel like this reward mechanism is very volatile. First, we're apparently "take()ing" the reward from the entry of the block. This means that, with no doubt, this block reward should be depleted completely by the end of this block. Using division like this potentially means we'll have remainder that will not be distributed. Besides, this will make the inflation schedule extremely chaotic. I don't see a reason for this. Simply give the rest of the reward to one of the authors. For simplicity, author number zero.

Second thing I have in mind, why do we have to use storage to determine the reward? The somehow glosses on the same question I had before. Why isn't the emission schedule clear and used?

Reply:
As per the discussion in the call, we reward all the fees of a given block to the block-author.